spec/facebook_ads/client_spec.rb

```diff
@@ -0,0 +1,216 @@
+# frozen_string_literal: true
+require 'spec_helper'
+
+describe FacebookAds::Client do
+  let(:access_token) { 'secret_token' }
+  let(:configuration) do
+    double(FacebookAds::Configuration, valid?: true, access_token: access_token)
+  end
+  let(:client) { described_class.new(configuration) }
+  let(:path) { 'test' }
+
+  let(:response) do
+    double(HTTParty::Response,
+           content_type: content_type,
+           body: response_body,
+           code: response_status)
+  end
+  let(:response_status) { 200 }
+  let(:content_type) { 'application/json' }
+  let(:response_body) { response_json.to_json }
+  let(:response_json) { Hash.new }
+
+  shared_examples_for 'performs an authenticated request' do |method|
+    it 'uses the access token as bearer token' do
+      auth_hash = { Authorization: "Bearer #{access_token}" }
+      headers_hash = { headers: auth_hash }
+      expect(HTTParty)
+        .to receive(method)
+        .with(anything, hash_including(headers_hash))
+        .and_return(response)
+
+      subject
+    end
+  end
+
+  shared_examples_for 'calls the versioned Graph API' do |method|
+    def expect_on_uri(method)
+      expect(HTTParty).to receive(method) do |uri, _|
+        yield URI(uri)
+        response
+      end
+    end
+
+    it 'uses the HTTPS scheme' do
+      expect_on_uri(method) { |uri| expect(uri.scheme).to eq 'https' }
+      subject
+    end
+
+    it 'calls the Graph API' do
+      expect_on_uri(method) do |uri|
+        expect(uri.host).to eq 'graph.facebook.com'
+      end
+      subject
+    end
+
+    it 'calls the versioned path' do
+      expect_on_uri(method) { |uri| expect(uri.path).to eq "/v2.7/#{path}" }
+      subject
+    end
+  end
+
+  shared_examples_for 'parses the response' do |method|
+    let(:response_json) { { 'test' => 'ok' } }
+
+    before do
+      allow(HTTParty)
+        .to receive(method)
+        .with(any_args)
+        .and_return(response)
+    end
+
+    it 'returns a parsed JSON hash' do
+      is_expected.to eql response_json
```

Comment:
When not used for a one-liner expectation I'd prefer the standard expect(subject).to eql response_json.